### PR TITLE
split bucket/prefix to get new bucket and prefix

### PR DESCRIPTION
**Pull request by KanShiori**
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
   ```yaml
    # ...
     gcs:
       projectId: xxxx
       secretName: gcs-secret
       bucket: test/123/123
       prefix: tidb-backup/xxx
     cleanPolicy: Delete
   ```
   ```yaml
    # ...
     s3:
       provider: aws
       secretName: tidb-s3-secret
       region: us-west-2
       bucket: test/123/123
       prefix: tidb-backup-2/xxx
   ```
* Backup: all backup is successful
* Restore: all retore is successful
* clean backup is successful
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: cmd/backup-manager/app/util/remote.go [hunk 1]
```diff
@@ -228,10 +228,13 @@ func newGcsStorageOption(conf *gcsConfig) []string {
 func makeS3Config(s3 *v1alpha1.S3StorageProvider, fakeRegion bool) *s3Config {
 	conf := s3Config{}
 
-	conf.bucket = s3.Bucket
+	path := strings.Trim(s3.Bucket, "/") + "/" + strings.Trim(s3.Prefix, "/")
```
Comment (csuzhangxc) [inline cmd/backup-manager/app/util/remote.go:231]: can we use `path.Join` here?

Comment (KanShiori) [reply cmd/backup-manager/app/util/remote.go:231]: `path.Join` will parse `../`, I don't know if it's safe, so I concatenate string.

Comment (DanielZhangQD) [reply cmd/backup-manager/app/util/remote.go:230]: I tried this on S3 and `..` is a valid name.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**